graylog2-server/src/main/java/org/graylog2/periodical/IndexRangesCleanupPeriodical.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.periodical;
+
+import com.github.joschi.jadconfig.util.Duration;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.eventbus.EventBus;
+import com.google.common.primitives.Ints;
+import com.google.inject.name.Named;
+import org.graylog2.indexer.Deflector;
+import org.graylog2.indexer.cluster.Cluster;
+import org.graylog2.indexer.esplugin.IndicesDeletedEvent;
+import org.graylog2.indexer.ranges.IndexRange;
+import org.graylog2.indexer.ranges.IndexRangeService;
+import org.graylog2.plugin.periodical.Periodical;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.SortedSet;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A {@link Periodical} to clean up stale index ranges (e. g. because the index has been deleted externally)
+ *
+ * @since 1.3.0
+ */
+public class IndexRangesCleanupPeriodical extends Periodical {
+    private static final Logger LOG = LoggerFactory.getLogger(IndexRangesCleanupPeriodical.class);
+
+    private final Cluster cluster;
+    private final Deflector deflector;
+    private final IndexRangeService indexRangeService;
+    private final EventBus eventBus;
+    private final int periodSeconds;
+
+    @Inject
+    public IndexRangesCleanupPeriodical(final Cluster cluster,
+                                        final Deflector deflector,
+                                        final IndexRangeService indexRangeService,
+                                        final EventBus eventBus,
```

Comment:
Only the Graylog master node manages the storage backend (removes indices etc.), so emitting this event into the cluster is unnecessary.